packages/block-editor/src/components/page-template-picker/use-page-template-picker.js

```diff
@@ -16,10 +23,7 @@ const __experimentalUsePageTemplatePickerVisible = () => {
 		const isOnlyUnmodifiedDefault =
 			blocks.length === 1 && isUnmodifiedDefaultBlock( firstBlock );
 		const isEmptyContent = isEmptyBlockList || isOnlyUnmodifiedDefault;
-		const isPage = getCurrentPostType() === 'page';
 
-		return isEmptyContent && isPage;
+		return isEmptyContent && isTemplatePickerAvailable;
```

Comment:
I'm not sure if I'm overthinking this, but I imagine the majority of times this runs it will return false. I'd suggest refactoring a bit to return early and avoid any extra calculations:

```jsx
if ( ! isTemplatePickerAvailable ) {
    return false;
}
//...
if ( isEmptyBlockList ) {
   return false;
}
// ...
```

It'd be interesting to check if this impacts performance at all, or I'm just worried about nothing 😅

Reply:
I'm always down for performance improvements 🙌, I'll definitely check it out, thanks for the review koke!